### PR TITLE
e2e: add test for luks encryted state disk

### PR DESCRIPTION
**Pull request by 3u13r**
This is a precursor to the end-user documentation on how to use the workloadsecret to mount a luks-encrypted CSI disk.

### CODE REVIEW

File: internal/kuberesource/sets.go [hunk 1]
```diff
@@ -496,3 +497,128 @@ func Emojivoto(smMode serviceMeshMode) []any {
...
+# workload_secret_path is the path to the Contrast workload secret.
+workload_secret_path="/contrast/secrets/workload-secret-seed"
+# tmp_key_path is the path to a temporary key file.
+tmp_key_path="/dev/shm/key"
+# disk_encryption_key_path is the path to the disk encryption key.
+disk_encryption_key_path="/dev/shm/disk-key"
+
+# If the device is already a LUKS device, format it.
+if ! cryptsetup isLuks "${device}"; then
+	# Generate a random key for the first initialization.
```
Comment (Freax13) [inline internal/kuberesource/sets.go:519]: Have you considered removing the key slot with this key after initialization?

File: justfile [hunk 2]
```diff
@@ -101,10 +104,11 @@ generate cli=default_cli platform=default_platform:
     nix run -L .#{{ cli }} -- generate \
         --workspace-dir ./{{ workspace_dir }} \
         --image-replacements ./{{ workspace_dir }}/just.containerlookup \
-        --reference-values {{ platform }}\
+        --reference-values {{ platform }} \
+        --genpolicy-cache-path ./{{ workspace_dir }}/../layers-cache.json \
         ./{{ workspace_dir }}/deployment/*.yml
 
-    # On baremetal SNP, we don't have default values for MinimumTCB, so we need to set some here.
+    # On baremetal SNP, we do not have default values for MinimumTCB, so we need to set some here.
```
Comment (Freax13) [inline justfile:111]: Why was this changed?

File: packages/by-name/microsoft/genpolicy/package.nix [hunk 3]
```diff
@@ -45,6 +45,8 @@ rustPlatform.buildRustPackage rec {
       #
       # This patch is not upstreamable.
       ./0004-genpolicy-regex-check-contrast-specific-layer-src-pr.patch
+      ./0005-genpolicy-propagate-mount_options-for-empty-dirs.patch
+      ./0006-genpolicy-support-HostToContainer-mount-propagation.patch
```
Comment (Freax13) [inline packages/by-name/microsoft/genpolicy/package.nix:49]: Can we add some comments here for the new patches?

File: packages/containers.nix [hunk 4]
```diff
@@ -78,6 +78,27 @@ let
...
+        bash
+        bashInteractive
+        coreutils
+        ncurses
+        procps
+        cryptsetup
+        e2fsprogs # mkfs.ext4
+        mount
+        util-linux # blkid
+        openssl
```
Comment (Freax13) [inline packages/containers.nix:94]: Are all of these needed?

File: packages/by-name/microsoft/genpolicy/package.nix (hunk 3 above)
Comment (burgerdev) [reply packages/by-name/microsoft/genpolicy/package.nix:49]: In particular what the upstream status is.

File: justfile (hunk 2 above)
Comment (3u13r) [reply justfile:111]: The `'` broke my syntax highlighting, so I changed it temporarily. Since it's unrelated I'll revert this.

File: justfile [hunk 5]
```diff
@@ -101,10 +104,11 @@ generate cli=default_cli platform=default_platform:
     nix run -L .#{{ cli }} -- generate \
         --workspace-dir ./{{ workspace_dir }} \
         --image-replacements ./{{ workspace_dir }}/just.containerlookup \
-        --reference-values {{ platform }}\
+        --reference-values {{ platform }} \
+        --genpolicy-cache-path ./{{ workspace_dir }}/../layers-cache.json \
```
[the remark this answers is not in the document]
Comment (3u13r) [reply justfile:108]: Ah sorry, I oversaw this. But generally we currenly save the layers-cache in the workspace folder, which we clean up on every `just default` command and I wanted some deployment speed. Since this is the justfile we might discuss doing such a change (in another PR) though.

File: e2e/volumestatefulset/volumestatefulset_test.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		pods, err := ct.Kubeclient.PodsFromOwner(ctx, ct.Namespace, "StatefulSet", "volume-tester")
+		require.NoError(err)
+		require.Len(pods, 1)
+
+		require.NoError(ct.Kubeclient.Restart(ctx, kubeclient.StatefulSet{}, ct.Namespace, "volume-tester"))
+		require.NoError(ct.Kubeclient.WaitFor(ctx, kubeclient.StatefulSet{}, ct.Namespace, "volume-tester"))
```
[the remark this answers is not in the document]
Comment (3u13r) [reply e2e/volumestatefulset/volumestatefulset_test.go:100]: Good (c/p)atch though. Thanks!

File: packages/containers.nix (hunk 4 above)
Comment (3u13r) [reply packages/containers.nix:94]: Combined the bash/shell related ones to busybox. Other than that, yes. I haven't measured it, but I tried not having to include openssl here, but I didn't find another option of doing hkdf "in a shell". We _might_ convert this to a Go program but we'd have to link againt libcryptsetup which is a bit of a pain in Constellation.

File: internal/kuberesource/sets.go (hunk 1 above)
Comment (3u13r) [reply internal/kuberesource/sets.go:519]: Ok that was easier than imagined. I thought I needed multiple commands to find out and then delete the key slot. Apparently not.

File: packages/containers.nix (hunk 4 above)
Comment (burgerdev) [reply packages/containers.nix]: The bash program does a better job at illustrating the ideas, though.

File: internal/kuberesource/sets.go (hunk 1 above)
Comment (burgerdev) [reply internal/kuberesource/sets.go:519]: Excuse my ignorance: what's the benefit of doing this, as opposed to initializing with the derived key right away?

**Review by burgerdev — APPROVED** [review]

File: internal/kuberesource/sets.go (hunk 1 above)
Comment (Freax13) [reply internal/kuberesource/sets.go:519]: We derive the key from the UUID of the LUKS partition and we don't know the UUID before we create it.

**Review by Freax13 — APPROVED** [review]
